src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/AutoCrafterListener.java

```diff
@@ -58,6 +65,21 @@ public void onInteract(PlayerRightClickEvent e) {
                     return;
                 }
 
+                // Check if the recipe of the item is disabled.
+                boolean unlocked = true;
+                for (Recipe recipe : Bukkit.getRecipesFor(e.getItem())) {
+                    if (!e.getPlayer().hasDiscoveredRecipe(((Keyed) recipe).getKey())) {
+                        unlocked = false;
+                        break;
+                    }
+                }
```

Comment:
This should be placed into a seperate method.

Reply:
This way you can also use an actual `return` statement, rather than this temp-variable + `break;` workaround

Reply:
Hmm... where should I declare the method then? In this same file?

Reply:
Also what should I annotate the method with...? I'm kinda confused with the @ annotations stuff...